frontend/src/model/beta/Releases.tsx

```diff
@@ -26,21 +32,27 @@ export default function Releases({ model }: { model: any }) {
             Draft new Release
           </Button>
         </Box>
+        {isReleasesLoading && <Typography sx={{ m: 4 }}>{`Fetching releases for model ${model.name}`}</Typography>}
+
+        {releases && releases.length === 0 && <EmptyBlob text={`No releases found for model ${model.name}`} />}
 
-        {model.releases.map((release) => {
-          return (
-            release.name && (
-              <ModelReleaseDisplay
-                key={release.semver}
-                modelId={model.id}
-                release={release}
-                latestRelease={latestRelease}
-              />
-            )
-          )
-        })}
+        {releases &&
+          releases
+            .sort((a, b) => (a.semver < b.semver ? 1 : b.semver > a.semver ? -1 : 0))
```

Comment:
This can be written in a more readable way and is also duplicating the same sorting logic. See my other comments for more info.